riscv/triggers.cc

```diff
@@ -177,6 +177,15 @@ bool module_t::tdata1_write(processor_t * const proc, unsigned index, const reg_
   reg_t tdata1 = val;
   reg_t tdata2 = triggers[index]->tdata2_read(proc);
 
+  // hardware must zero chain in writes that set dmode to 0 if the next trigger has dmode of 1
+  if (index+1 < triggers.size())
+    if (triggers[index+1]->get_dmode())
+      if (!get_field(val, CSR_TDATA1_DMODE(xlen)))
+        if (type == 2 || type == 6) {
```

Comment:
Don't use magic numbers.

This doesn't need to be 4 nested `if` statements, it's just a compound expression (`&&`).

I don't think this is the right place to do this. It should be down inside `mcontrol_t` where the `chain` bit is implemented -- but that doesn't have enough information. Add a `bool allow_chain` parameter to `trigger_t::tdata1_write()`?